src/fs/inotify.rs

```diff
@@ -0,0 +1,43 @@
+//! inotify support for working with inotifies
+
+pub use crate::backend::fs::inotify::{CreateFlags, WatchFlags};
+use crate::backend::fs::syscalls;
+use crate::fd::{AsFd, OwnedFd};
+use crate::io;
+
+/// `inotify_init1(flags)`—Creates a new inotify object.
+///
+/// Use the [`CreateFlags::CLOEXEC`] flag to prevent the resulting file
+/// descriptor from being implicitly passed across `exec` boundaries.
+#[doc(alias = "inotify_init1")]
+#[inline]
+pub fn inotify_init(flags: CreateFlags) -> io::Result<OwnedFd> {
+    syscalls::inotify_init1(flags)
+}
+
+/// `inotify_add_watch(self, path, flags)`—Adds a watch to inotify.
+///
+/// This registers or updates a watch for the filesystem path `path` and
+/// returns a watch descriptor corresponding to this watch.
+///
+/// Note: Due to the existence of hardlinks, providing two different paths to
+/// this method may result in it returning the same watch descriptor. An
+/// application should keep track of this externally to avoid logic errors.
+#[inline]
+pub fn inotify_add_watch<P: crate::path::Arg>(
+    inot: impl AsFd,
```

Comment:
Could you add the TODO comments about changing this `impl AsFd` to a generic argument in the future?

Reply:
See https://github.com/bytecodealliance/rustix/issues/753#issuecomment-2291869835. I feel like it'll be more effective to run a global pass on the whole lib rather than trying to remember the spots impls need to be fixed. If you disagree, I'll be happy to add the TODOs here.

Reply:
Ah, makes sense.